sklearn/utils/sparsefuncs_fast.pyx

```diff
@@ -14,12 +15,109 @@ cimport cython
 
 np.import_array()
 
-
 ctypedef np.float64_t DOUBLE
 
 @cython.boundscheck(False)
 @cython.wraparound(False)
 @cython.cdivision(True)
+def csc_min_axis0(X):
+    """Minimum of each element of X down axis 0.
+    Equivalent to calling X.min(axis=0)
+    """
+
+    cdef:
+        unsigned int n_features = X.shape[1]
+        int[::1] col_ptr = X.indptr
+        np.ndarray[DOUBLE, ndim=1, mode="c"] minimum = np.zeros(n_features)
+        np.ndarray[DOUBLE, ndim=1, mode="c"] data = X.data
+
+        unsigned int i, j
+
+    for i in range(n_features):
+        start = col_ptr[i]
+        end = col_ptr[i + 1]
+        # Iterate downwards through matrix
+        for j in range(start, end):
+            minimum[i] = double_min(minimum[i], data[j])
+
+    return minimum
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+@cython.cdivision(True)
+def csc_max_axis0(X):
+    """Minimum of each element of X down axis 0.
+    Equivalent to calling X.min(axis=0)
+    """
+
+    cdef:
+        int n_features = X.shape[1]
+        col_ptr = X.indptr
+        np.ndarray[DOUBLE, ndim=1, mode="c"] maximum = np.zeros(n_features)
+        np.ndarray[DOUBLE, ndim=1, mode="c"] data = X.data
+
+        unsigned int i, j
+
+    for i in range(n_features):
+        start = col_ptr[i]
+        end = col_ptr[i + 1]
+
+        # Iterate downwards through matrix
+        for j in range(start, end):
+            maximum[i] = double_max(maximum[i], data[j])
+
+    return maximum
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+@cython.cdivision(True)
+def csr_min_axis0(X):
+    """Minimum of each element of X down axis 0.
+    Equivalent to calling X.min(axis=0)
+    """
+    cdef:
+        int n_features = X.shape[1]
+        col_indices = X.indices
+        np.ndarray[DOUBLE, ndim=1, mode="c"] minimum = np.zeros(n_features)
+        np.ndarray[DOUBLE, ndim=1, mode="c"] data = X.data
+        #DOUBLE[::1] data = X.data
```

Comment:
When I used `DOUBLE[::1] data = X.data` instead of `np.ndarray[DOUBLE, ndim=1, mode="c"]`, I ended up getting segmentation faults when I ran my code while running `nosetests test_sparsefuncs.py`. Can someone help explain why?